vector/src/main/java/im/vector/activity/VectorHomeActivity.java

```diff
@@ -638,48 +638,66 @@ private void checkNotificationPrivacySetting() {
+                            .setMessage(R.string.startup_notification_privacy_message)
+                            .setPositiveButton(R.string.startup_notification_privacy_button_grant, new DialogInterface.OnClickListener() {
+                                @Override
+                                public void onClick(DialogInterface dialog, int which) {
+                                    Log.d(LOG_TAG, "checkNotificationPrivacySetting: user wants to grant the IgnoreBatteryOptimizations permission");
+
+                                    // Request the battery optimization cancellation to the user
+                                    SystemUtilsKt.requestDisablingBatteryOptimization(VectorHomeActivity.this,
+                                            RequestCodesKt.BATTERY_OPTIMIZATION_REQUEST_CODE);
+                                }
+                            })
+                            .setNegativeButton(R.string.startup_notification_privacy_button_other, new DialogInterface.OnClickListener() {
+                                @Override
+                                public void onClick(DialogInterface dialog, int which) {
+                                    Log.d(LOG_TAG, "checkNotificationPrivacySetting: user opens notification policy setting screen");
+
+                                    // open the notification policy setting screen
+                                    startActivity(NotificationPrivacyActivity.getIntent(VectorHomeActivity.this));
+                                }
+                            })
+                            .show();
+                }
+            }
+        } else {
+            if (!PreferencesManager.didAskUserToIgnoreBatteryOptimizations(this)) {
+                PreferencesManager.setDidAskUserToIgnoreBatteryOptimizations(this);
+
+                if (!SystemUtilsKt.isIgnoringBatteryOptimizations(this)) {
+                    new AlertDialog.Builder(this)
+                            .setCancelable(false)
+                            .setTitle(R.string.startup_notification_fdroid_battery_optim_title)
+                            .setMessage(R.string.startup_notification_fdroid_battery_optim_message)
+                            .setPositiveButton(R.string.startup_notification_fdroid_battery_optim_button_grant, new DialogInterface.OnClickListener() {
+                                @Override
+                                public void onClick(DialogInterface dialog, int which) {
+                                    Log.d(LOG_TAG, "checkNotificationPrivacySetting: user wants to grant the IgnoreBatteryOptimizations permission");
+
+                                    // Request the battery optimization cancellation to the user
+                                    SystemUtilsKt.requestDisablingBatteryOptimization(VectorHomeActivity.this,
+                                            RequestCodesKt.BATTERY_OPTIMIZATION_REQUEST_CODE);
```

Comment:
Is the code in onActivityResult() also applicable for F-Droid? I do not think so, so may be consider using another request code